<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Eleanor Zhang" />

<meta name="date" content="2022-09-19" />

<title>FactorGO</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="ZixuanZhang_resume.pdf">Resume</a>
</li>
<li>
  <a href="FactorGo.html">FactorGO</a>
</li>
<li>
  <a href="RaDtest.html">RaDtest</a>
</li>
<li>
  <a href="Gap_model.html">Gap model</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="mailto:&lt;zzhang39@usc.edu&gt;">
    <span class="fa fa-envelope fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://www.linkedin.com/in/zixuan-z-65a769121/">
    <span class="fa fa-linkedin"></span>
     
  </a>
</li>
<li>
  <a href="http://github.com/zixuanzhang">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/elezzx2020">
    <span class="fa fa-fw fa-twitter-square"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">FactorGO</h1>
<h4 class="author">Eleanor Zhang</h4>
<h4 class="date">2022-09-19</h4>

</div>


<div id="background" class="section level2">
<h2>Background</h2>
<ul>
<li>Biobank genome-wide association studies (<a
href="https://pubmed.ncbi.nlm.nih.gov/28686856/">GWAS</a>) identified
thousands of genetic variants associated with human phenotypic
traits.<br />
</li>
<li>Pleiotropy is pervasive in genetic variants (one variant affects
multiple traits) and thus provide opportunities learn shared mechanisms
among human phenotypic traits through the lens of shared genetic
associations.<br />
</li>
<li><a
href="https://www.nature.com/articles/s41467-019-11953-9">Previous
work</a> proposed truncated singular-value decomposition (tSVD) on
multiple GWAS summary data to characterize shared latent genetic
pleiotropic factors, which uses effect sizes from GWAS significant
variants without modeling effect size uncertainty.</li>
<li>We propose <strong>FactorGO</strong>, a factor analysis model on
genetic association to account for uncertainty in effect-size estimates
and leverage an automatic relevance determination (ARD) prior to control
sparsity on factor loadings.</li>
</ul>
</div>
<div id="study-overview" class="section level2">
<h2>Study overview</h2>
<p>We constructed data matrix of Z score summmary statistics for 51,399
genetic variants among 2,483 traits from <a
href="https://pan.ukbb.broadinstitute.org/">Pan-UK Biobank</a>. After
inferring K=100 latent factors, we interpret each factor by:<br />
1) Fig b: quantify relative importance of trait within a factor by
contribution score, as described <a
href="https://www.nature.com/articles/s41467-019-11953-9">elsewhere</a>.<br />
2) Fig c: test for enrichment of cell type specific functional
annotation using <a href="https://github.com/bulik/ldsc">LDSC</a>.</p>
<p><img src = "images/FactorGO_Diagram.png" style = "width:70%" align = "middle"></p>
</div>
<div id="result" class="section level2">
<h2>Result</h2>
<div id="runtime-comparison" class="section level3">
<h3>Runtime comparison</h3>
<p>FactorGO implemented using Just-In-Time (<a
href="https://jax.readthedocs.io/en/latest/jax-101/02-jitting.html">JIT</a>)
compilation via JAX package in python under a parameter expansion method
substantially improves convergence rate, which make its run time on
2-cores GPU comparable to tSVD:</p>
<p><img src = "images/runtime.png" width="50%" height="50%" align = "middle"></p>
</div>
<div id="top-two-factors" class="section level3">
<h3>Top two factors</h3>
<p>To look at the pleiotropic factors, we start with top 2 factors that
explains most variance in the data. The first factor characterize shared
co-morbidity between <a
href="https://www.cdc.gov/mmwr/volumes/70/wr/mm7010e4.htm">Covid-19 and
body weight</a>, basal metabolic rate that is relevant with BMI. The
second factor characterize overall human growth such as height, artery
measure, and body fat volume.</p>
<p><img src = "images/FactorGO_Top2_Z1_Z2.png" width="50%" height="50%" align = "middle"></p>
<p>*highlighted points are flanked by 2 times posterior variance</p>
</div>
<div id="case-study-1-human-height" class="section level3">
<h3>Case study 1: Human height</h3>
<p>For standing height, the most relevant pleiotropic factor is factor
2. In this factor, the leading traits are body measures such as standing
height, sitting height, followed by fat-free mass measurements. These
reflect the growth hormones play roles in both height growth and
metabolic pathways. Moreover, this factor is enriched with genes
differentially expressed in reproductive organs like Uterus and Cervix
and muscular skeletal tissues like cartilage and chondrocytes , which is
consistent with previous findings (<a
href="https://www.nature.com/articles/ng.3097">Wood et al. 2014</a>, <a
href="https://www.nature.com/articles/s41588-018-0081-4">Finucane et
al. 2018</a>).</p>
<p><img src = "images/FactorGO_height_ldsc_F2_SEG.png" width="80%" height="80%" align = "middle"></p>
</div>
<div id="case-study-2-rheumatoid-arthritis-ra" class="section level3">
<h3>Case study 2: Rheumatoid arthritis (RA)</h3>
<p>For RA, we find the most relevant factor captures quite a few trait
domain including bone related traits like blood calcium level,
osteoporosis (easiness of bone fracture); a known treatment for RA
called methotrexate; and moreover, other autoimmune disease like IBD.
Interestingly, the variants driving this factor is enriched with genes
that are differentially expressed in periodontal condition and digestive
system, which also reflect existing clinical evidence (<a
href="https://pubmed.ncbi.nlm.nih.gov/27974664/">Konig et
al. 2016</a>).</p>
<p><img src = "images/FactorGO_RA_ldsc_F86_SEG.png" width="80%" height="80%" align = "middle"></p>
</div>
</div>
<div id="conclusion" class="section level2">
<h2>Conclusion</h2>
<ul>
<li>FactorGO identifies pleiotropic factors while appropriately
accounting for uncertainty in effect sizes under probabilistic
model.<br />
</li>
<li>We applied FactorGO on 2,438 traits and 205 functional annotations
to characterize pleiotropic factors.<br />
</li>
<li>FactorGO validated:<br />
</li>
</ul>
<ol style="list-style-type: decimal">
<li>shared co-morbidity between Covid-19 and BMI.<br />
</li>
<li>shared hormonal effect in height and body fat measures.<br />
</li>
<li>shared inflammatory mechanisms for rheumatoid arthritis.</li>
</ol>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
