<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Eleanor Zhang" />

<meta name="date" content="2022-10-01" />

<title>RaDtest</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="CV.html">CV</a>
</li>
<li>
  <a href="FactorGo.html">FactorGO</a>
</li>
<li>
  <a href="RaDtest.html">RaDtest</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="mailto:&lt;zzhang39@usc.edu&gt;">
    <span class="fa fa-envelope fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://www.linkedin.com/in/zixuan-z-65a769121/">
    <span class="fa fa-linkedin"></span>
     
  </a>
</li>
<li>
  <a href="http://github.com/zixuanzhang">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/elezzx2020">
    <span class="fa fa-fw fa-twitter-square"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">RaDtest</h1>
<h4 class="author">Eleanor Zhang</h4>
<h4 class="date">2022-10-01</h4>

</div>


<div id="introduction" class="section level2">
<h2>Introduction</h2>
<ul>
<li>Rare genetic variants with allele frequency lower than 1% or 5% can
have profound impact on protein structure, but exist in low frequency in
nature due to purifying selection.<br />
</li>
<li>Variants in a true risk gene can have varying deleterious impact.
The most common type of variant in protein coding region is due to
missense mutation.<br />
</li>
<li>We propose a variable threshold association test called RaDtest,
which uses in silico predicted deleteriousness (REVEL score) as weight
and find gene-specific optimal weighting function in association
analysis.<br />
</li>
<li>I conducted simulation study to validate type I error rate and power
analysis using real data of rare genetic variants in BMPR2 in the
context of pulmonary heart disease (PAH), which is a very rare disease
across all ages.</li>
</ul>
</div>
<div id="method" class="section level2">
<h2>Method</h2>
<div id="gene-specific-statistical-test" class="section level3">
<h3>Gene-specific statistical test</h3>
<p>The goal is to test whether a gene is associated with a disease
phenotype, based on missense genetic variants in this gene. For a
dichotomous disease phenotype, we define relative risk (RR) as the
occurrence of missense variants among cases versus that in control
group.</p>
<p><span class="math display">\[
\begin{align}
H0: RR = 1 \\
H1: RR &gt; 1
\end{align}
\]</span> We assume that the relative risk of a missense variant in a
true risk gene is correlated with <em>in silico</em> predicted
deleteriousness (prediced by REVEL score) and this relationship can be
captured by a sigmoid function (soft threshold). The weight of each
missense variants for a given region or a gene by their in silico
prediction (x) in a sigmoid function f(x) = L/(1 + exp(- K * (x - x0)),
and then use logistic regression to test the association. When K is
sufficiently large, the sigmoid weight function changes abruptly at
midpoint such that it is equivalent to a hard threshold. For example in
the plot below, for a genetic variant with REVEL score 0.5 (dashed) and
score 0.9 (solid), the weight function gets closer to hard threshold
when K increases.</p>
<p><img src = "images/sigmoid_weight.png" width="50%" height="50%" align = "middle"></p>
<p>For a given gene, RaDtest search for an optimal threshold <span
class="math inline">\(x_0\)</span> for REVEL score where the maximal
significance in case-control comparison is reached with smallest p-value
<span class="math inline">\(ps_{0}\)</span>. Statistical significance is
evaluated by permuting class labels of cases and controls. In each
permutation, we repeat the searching of <span
class="math inline">\(x_0\)</span>  and record the smallest p value
<span class="math inline">\(ps_{i}\)</span>. We define empirical p value
as <span class="math inline">\((Np+1)/(N+1)\)</span>, where Np is the
number of permutations for which <span
class="math inline">\(ps_{i}\)</span> is smaller than <span
class="math inline">\(ps_{0}\)</span> out of total N permutation.</p>
</div>
<div id="data-for-simulation" class="section level3">
<h3>Data for simulation</h3>
<p>We collected 93 missense BMPR2 genetic variants with population
allele frequency less than 0.01% from 3100 patients with PAH in recent
publication (Machado 2015), and additional 31 missense variants from
7718 controls from gnomAD European WGS database. This data exempted us
from making any assumption about the relationship of genetic effect and
REVEL score. We benchmarked the performance of RaDtest under varying
effect size (RR) and gene length (cumulative minor allele frequency) by
simulated realistic data. </p>
</div>
</div>
<div id="result" class="section level2">
<h2>Result</h2>
<div id="validate-type-i-error" class="section level3">
<h3>Validate Type I error</h3>
<p>By simulating 10,000,000 data under null model, where samples are
pooled and randomly relabeled as either case or control. Under different
<span class="math inline">\(\alpha\)</span> threshold, we showed the
type I error rate is smaller than the expected type I error rate.</p>
<table class=" lightable-classic" style="font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Expected and observed type I error rate
</caption>
<thead>
<tr>
<th style="text-align:right;">
alpha theshold
</th>
<th style="text-align:left;">
observed Type I error rate
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
1.0e-03
</td>
<td style="text-align:left;">
9.85e-04
</td>
</tr>
<tr>
<td style="text-align:right;">
1.0e-04
</td>
<td style="text-align:left;">
9.93e-05
</td>
</tr>
<tr>
<td style="text-align:right;">
1.0e-05
</td>
<td style="text-align:left;">
9.80e-06
</td>
</tr>
<tr>
<td style="text-align:right;">
2.5e-06
</td>
<td style="text-align:left;">
2.10e-06
</td>
</tr>
</tbody>
</table>
</div>
<div id="power-analysis" class="section level3">
<h3>Power analysis</h3>
<p>We explore power of RaDtest under<br />
1) varying relative risk and <span class="math inline">\(\alpha\)</span>
threshold<br />
2) different gene length (longer gene tends to have more genetic
missense variants)</p>
<p>As a result, we observe greater fraction of true cases and less
stringent alpha threshold increases power in Figure A. Increased gene
length and thus increased minor cumulative allele frequency (CMAF) can
also increase statistical power, comparing two scenarios under 60% and
70% true cases. Power increases as more variants are included in the
association test.</p>
<div class="figure" style="text-align: center">
<img src="images/power1.png" alt="caption" width="49%" height="20%" /><img src="images/power2.png" alt="caption" width="49%" height="20%" />
<p class="caption">
caption
</p>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
