---
title: "RaDtest"
author: "Eleanor Zhang"
date: '2022-10-01'
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(kableExtra)
```

## Introduction

* Rare genetic variants with allele frequency lower than 1% or 5% can have profound impact on protein structure, but exist in low frequency in nature due to purifying selection.  
* Variants in a true risk gene can have varying deleterious impact. The most common type of variant in protein coding region is due to missense mutation.  
* We propose a variable threshold association test called RaDtest, which uses in silico predicted deleteriousness (REVEL score) as weight and find gene-specific optimal weighting function in association analysis.  
* I conducted simulation study to validate type I error rate and power analysis using real data of rare genetic variants in BMPR2 in the context of pulmonary heart disease (PAH), which is a very rare disease across all ages. 

## Method 
### Gene-specific statistical test

The goal is to test whether a gene is associated with a disease phenotype, based on missense genetic variants in this gene. For a dichotomous disease phenotype, we define relative risk (RR) as the occurrence of missense variants among cases versus that in control group.

$$
\begin{align}
H0: RR = 1 \\
H1: RR > 1
\end{align}
$$
We assume that the relative risk of a missense variant in a true risk gene is correlated with _in silico_ predicted deleteriousness (prediced by REVEL score) and this relationship can be captured by a sigmoid function (soft threshold). The weight of each missense variants for a given region or a gene by their in silico prediction (x) in a sigmoid function f(x) = L/(1 + exp(- K * (x - x0)), and then use logistic regression to test the association. When K is sufficiently large, the sigmoid weight function changes abruptly at midpoint such that it is equivalent to a hard threshold. For example in the plot below, for a genetic variant with REVEL score 0.5 (dashed) and score 0.9 (solid), the weight function gets closer to hard threshold when K increases.

<img src = "images/sigmoid_weight.png" width="50%" height="50%" align = "middle">


For a given gene, RaDtest search for an optimal threshold $x_0$ for REVEL score where the maximal significance in case-control comparison is reached with smallest p-value $ps_{0}$. Statistical significance is evaluated by permuting class labels of cases and controls. In each permutation, we repeat the searching of $x_0$  and record the smallest p value $ps_{i}$. We define empirical p value as $(Np+1)/(N+1)$, where Np is the number of permutations for which $ps_{i}$ is smaller than $ps_{0}$ out of total N permutation.

### Data for simulation

We collected 93 missense BMPR2 genetic variants with population allele frequency less than 0.01% from 3100 patients with PAH in recent publication (Machado 2015), and additional 31 missense variants from 7718 controls from gnomAD European WGS database. This data exempted us from making any assumption about the relationship of genetic effect and REVEL score. We benchmarked the performance of RaDtest under varying effect size (RR) and gene length (cumulative minor allele frequency) by simulated realistic data. 

## Result

### Validate Type I error 

By simulating 10,000,000 data under null model, where samples are pooled and randomly relabeled as either case or control. Under different $\alpha$ threshold, we showed the type I error rate is smaller than the expected type I error rate. 

```{r echo=FALSE}
tibble(alpha = c(1e-3, 1e-4, 1e-5, 2.5e-6),
       type_I_error = format(c(9.85e-4, 9.93e-5, 9.80e-6, 2.10e-6), digits=3)) %>% 
  kbl(caption = "Expected and observed type I error rate",
      col.names=c('alpha theshold', 'observed Type I error rate')) %>%
  kable_classic(full_width = F, html_font = "Cambria")
```

### Power analysis

We explore power of RaDtest under  
1) varying relative risk and $\alpha$ threshold  
2) different gene length (longer gene tends to have more genetic missense variants)

As a result, we observe greater fraction of true cases and less stringent alpha threshold increases power in Figure A. Increased gene length and thus increased minor cumulative allele frequency (CMAF) can also increase statistical power, comparing two scenarios under 60% and 70% true cases. Power increases as more variants are included in the association test.

```{r, echo=FALSE,out.width="49%",out.height="20%",fig.cap="caption",fig.show='hold',fig.align='center'}
knitr::include_graphics(c("images/power1.png","images/power2.png"))
``` 

